src/util/virutil.c

```diff
@@ -277,6 +277,67 @@ virParseVersionString(const char *str, unsigned long *version,
     return 0;
 }
 
+int
+virCHVersionString(const char *str, unsigned long *version,
+                      bool allowMissing)
+{
+    int ret = virParseVersionString(str, &version, true);
+    if ( ret < 0){
+        unsigned int major, minor = 0, micro = 0;
+
+        if (strncmp(str, "cloud-hypervisor v", 18) == 0) {
+            /**
+             * Check for regular 'cloud-hypervisor --version' output with major,minor and micro version
+             * This format is for upstream CLH release version
+             * For ex: cloud-hypervisor v32.0.0
+             */
+            if(sscanf(str, "cloud-hypervisor v%d.%d.%d", &major, &minor, &micro) != 3){
+                /**
+                 * Try extracting major and minor version only
+                 * This format is for upstream CLH main branch build
+                 * For ex: cloud-hypervisor v33.0-104-ge0e3779e-dirty
+                 */
+                if(sscanf(str, "cloud-hypervisor v%d.%d", &major, &minor) != 2){
+
+                    /**
+                     * Log Error if none of the format matches and return -1
+                     */
+                    fprintf(stderr, "Can not extract upstream CLH version: %s\n", str);
+                    return -1;
+                }
+            }
+        } else {
+            /**
```

Comment:
I'm not sure to understand. If `virParseVersionString` call didn't fail, the result can be just returned?

Thanks

Reply:
This is modified with latest push. Please check